Automattic/wp-calypso
# Make PeopleProfile a function component and use react-query to fetch external contributors

#### Changes proposed in this Pull Request
I'm hoping to use react-query on the `PeopleProfile` component for another PR I'm working on, and I needed the component to be a functional one for that reason. I don't want to do too much in one PR so it becomes unmanageable, so this PR makes the component functional and introduces a react-query version that gets external contributors. 

Note that there are still other components using the existing `requestExternalContributors` data-getter. It became a little hairy to change them all in one go, so I'm starting with this one. It's not ideal to have both, but it is a good way to make incremental changes. The external contributors call is a good candidate for react-query I think.

This PR also fixes a warning thrown on all `people/team` pages. It mostly looks something like this: 
`Warning: Cannot update a component ('Connect(Localized(PeopleListSectionHeader))') while rendering a different component ('Connect(Localized(WithLocalizedMoment(PeopleProfile)))'). To locate the bad setState() call inside 'Connect(Localized(WithLocalizedMoment(PeopleProfile)))', follow the stack trace as described in https://fb.me/setstate-in-render
    in Connect(Localized(WithLocalizedMoment(PeopleProfile))) (created by PeopleListItem)
...
`

#### Testing instructions
- Verify that the warning mentioned above is gone
- Check that the people listing on `people/team` is behaving like it always has.
